providers/default.rb

```diff
@@ -346,6 +346,16 @@ def execute_install_script(install_script)
       cleanup
     end
   rescue SystemExit
+    # sysvinit won't restart chef-client after we exit, use cron to do so
+    if node['init_package'] == 'init' && ::File.exist?('/etc/init.d/chef-client')
```

Comment:
this is likely brittle.  i think i'd prefer tighter integration with chef chef-client cookbook node attributes.  if people do the right thing with the chef-client cookbook to manage their config, then i think we can automatically do the thing here if they're using sysv init scripts.

otherwise, i think if people are using sysv init scripts without the chef-client cookbook that they should have to explicitly set a node attribute in this cookbook to invoke this.

Reply:
You mean that if instead if `node['chef_client']['init_style'] == "init"` then we can better trust that we can trust /etc/init.d/chef-client to be there and behave, and a `node['chef_client_updater']['restart_via_cron']` attribute as an alternative for everyone else?